### PR TITLE
feat: Frontend map utils

### PR DESCRIPTION
**Pull request by Norbiros**

### CODE REVIEW

**Comment by goteusz-maszyk** [comment]
Imagine using leaflet
https://maplibre.org/

**Comment by Norbiros** [comment]
How using maplibre I can eaisly draw a polygon, display overlay image/layer, draw point etc?

**Comment by wikipop** [comment]
Test image display with [this](https://planetarycomputer.microsoft.com/api/data/v1/item/preview.png?collection=landsat-c2-l2&item=LC09_L2SP_182046_20240920_02_T1&assets=red&assets=green&assets=blue&color_formula=gamma+RGB+2.7%2C+saturation+1.5%2C+sigmoidal+RGB+15+0.55&format=png) before merging

File: frontend/pages/map.client.vue [hunk 1]
```diff
@@ -1,8 +1,16 @@
...
+  <Map :selected-polygon="polygon" :image-overlay="image" @map-click="mapClick" />
 </template>
 
-<script setup>
+<script setup lang="ts">
+import { LatLng } from "leaflet";
+
+const polygon: LatLng[] = [new LatLng(0, 0), new LatLng(1, 1), new LatLng(5000, 500)];
+const image = {
+  url: "https://www.printablee.com/postpic/2011/06/blank-100-square-grid-paper_405041.jpg",
```
Comment (wikipop) [inline frontend/pages/map.client.vue:10]: does it work with geocoded images? Read my other comment.

**Review by wikipop — CHANGES_REQUESTED** [review]

**Review by wikipop — APPROVED** [review]

**Pull request merged**